openpype/hosts/equalizer/api/host.py

```diff
@@ -0,0 +1,188 @@
+"""3dequalizer host implementation.
+
+note:
+    3dequalizer 7.1v2 uses Python 3.7.9
+
+"""
+import json
+import os
+import re
+
+import pyblish.api
+import tde4  # noqa: F401
+from attrs import asdict
+from attrs.exceptions import NotAnAttrsClassError
+from qtpy import QtCore, QtWidgets
+
+from openpype.host import HostBase, ILoadHost, IPublishHost, IWorkfileHost
+from openpype.hosts.equalizer import EQUALIZER_HOST_DIR
+from openpype.hosts.equalizer.api.pipeline import Container
+from openpype.pipeline import (
+    register_creator_plugin_path,
+    register_loader_plugin_path,
+)
+from openpype.tools.utils import get_openpype_qt_app
+
+CONTEXT_REGEX = re.compile(
+    r"AYON_CONTEXT::(?P<context>.*?)::AYON_CONTEXT_END",
+    re.DOTALL)
+PLUGINS_DIR = os.path.join(EQUALIZER_HOST_DIR, "plugins")
+PUBLISH_PATH = os.path.join(PLUGINS_DIR, "publish")
+LOAD_PATH = os.path.join(PLUGINS_DIR, "load")
+CREATE_PATH = os.path.join(PLUGINS_DIR, "create")
+INVENTORY_PATH = os.path.join(PLUGINS_DIR, "inventory")
+
+
+class EqualizerHost(HostBase, IWorkfileHost, ILoadHost, IPublishHost):
+    name = "equalizer"
+    _instance = None
+
+    def __new__(cls):
+        if not hasattr(cls, "_instance") or not cls._instance:
+            cls._instance = super(EqualizerHost, cls).__new__(cls)
+        return cls._instance
+
+    def __init__(self):
+        self._qapp = None
+        super(EqualizerHost, self).__init__()
+
+    def workfile_has_unsaved_changes(self):
+        """Return the state of the current workfile.
+
+        3DEqualizer returns state as 1 or zero, so we need to invert it.
+
+        Returns:
+            bool: True if the current workfile has unsaved changes.
+        """
+        return not bool(tde4.isProjectUpToDate())
+
+    def get_workfile_extensions(self):
+        return [".3de"]
+
+    def save_workfile(self, dst_path=None):
+        if not dst_path:
+            dst_path = tde4.getProjectPath()
+        result = tde4.saveProject(dst_path, True)
+        if not bool(result):
```

Comment:
This evaluates the same as just `if not result:` - why is the bool conversion needed?

Reply:
because `saveProject()` does not return bool. An yes, it is returnin 1 and 0 on success and python will handle that just as fine, but I like it more explicit.